scipy/sparse/csgraph/tests/test_shortest_path.py

```diff
@@ -98,6 +120,26 @@ def check(method, directed_in):
         for directed_in in (True, False):
             check(method, directed_in)
 
+def test_directed_sparse_zero():
+    # test directed sparse graph with zero-weight edge and two connected components
+    def check(method):
+        SP = shortest_path(directed_sparse_zero_G, method=method, directed=True,
+                           overwrite=False)
+        assert_array_almost_equal(SP, directed_sparse_zero_SP)
+
+    for method in methods:
+        check(method)
+
+def test_undirected_sparse_zero():
```

Comment:
could condense to  a single test parametrized over `directed_sparse_zero_SP, directed_sparse_zero_SP` and `methods`. Can't fault you for using the same test structure as elsewhere in the module though